bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ClientLibrariesImpl.java

```diff
@@ -272,59 +282,62 @@ private String getInline(LibraryType libraryType) {
      *
      * @return {@link Set<String>} of clientlib categories
      */
+    @NotNull
     protected Set<String> getCategoriesFromComponents() {
-        Set<String> categories = new HashSet<>();
```

Comment:
Here also I would prefer keeping the former implementation and not using the Java stream, which I find harder to debug and harder to follow with a recursion.